### PR TITLE
GH-2: remove `retention_days` param for the diagnostic-settings module

### PR DESCRIPTION
**Pull request by taoufik07**
Hey 🤗 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Changes proposed in this pull request

- Remove `retention_days` param for the diagnostic-settings module

@claranet/fr-azure-reviewers

### CODE REVIEW

File: r-logs.tf [hunk 1]
```diff
@@ -1,13 +1,12 @@
 module "diagnostics" {
   source  = "claranet/diagnostic-settings/azurerm"
-  version = "6.2.0"
+  version = "6.5.0"
```
Comment (BzSpi) [inline r-logs.tf:3]:
```suggestion
  version = "~> 6.5.0"
```
We've started to use this notation for all included modules to have fixes propagated quickly.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,19 @@
+# Unreleased
+
+Breaking
+  * GH-1: remove `retention_days` parameter for the diagnostic-settings module
```
Comment (Shr3ps) [inline CHANGELOG.md:4]:
```suggestion
  * [GH-2](https://github.com/claranet/terraform-azurerm-db-postgresql-flexible/pull/2/files): Remove `retention_days` parameter for the diagnostic-settings module
```

**Review by Shr3ps — CHANGES_REQUESTED** [review]

File: r-logs.tf (hunk 1 above)
Comment (taoufik07) [reply r-logs.tf]: done!

File: CHANGELOG.md (hunk 2 above)
Comment (taoufik07) [reply CHANGELOG.md]: done

**Comment by Shr3ps** [comment]
Can you rebase and fix CI?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,19 @@
+# Unreleased
```
Comment (taoufik07) [inline CHANGELOG.md:1]: Is `pre-commit` run in CI? When I run it locally on all files `pre-commit run --all-files`,  `end-of-file-fixer` fails to pass

Comment (Shr3ps) [reply CHANGELOG.md:1]: Oh yes, fix-end-of-line is not in CI, but you can push the fixes if you want :)
(the main reason is that we introduced pre-commit commit after the CI)

**Review by Shr3ps — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (taoufik07) [reply CHANGELOG.md:1]: done!

**Review by rossifumax — APPROVED** [review]
Thanks for your contrib !

**Pull request merged**